SQL Optimization

Helmphlyx opened the pull request:
## Describe your changes
Major Changes:
1. Refactor SQL code into modularized functions to allow testability
2. Parameterize SQL code instead of string formatting to handle safe SQL payloads
3. Fix database check so data is not reinitialized if database exists

Minor Changes:
1. Viewable notes for exercises in a workout
2. Add delete workout button when editing a workout
3. Search functionality in search exercises page

## Checklist before requesting a review
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas

Helmphlyx committed:
fix: modularize sql calls, add delete workout button, check for unique exercise/workout names

Helmphlyx committed:
fix: database check

Helmphlyx committed:
refactor sql

Helmphlyx committed:
add search utility to search exercises page

Helmphlyx committed:
add viewable for exercise notes

Helmphlyx committed:
add flash message for duplicate exercises

Helmphlyx committed:
remove if condition for workout search

Helmphlyx committed:
update docstrings

Taina-conde commented:
I liked what you did with the SQL, it's more testable and clear now. 
I was testing the app manually and found some inconsistencies. 
1. When searching for an exercise name, if the exercise was added by us, it sanitizes the input and finds the exercise. However, if I search for exercises names that came from the API, I need to type the capital letters or lower letters exactly like in the API. It would be helpful to sanitize the names that come from the API if possible. 
2. If I press the delete button in the workout and the exercise, it is not actually deleting them, they still appear in the search page. 
3. When editing a workout, if I delete an exercise, I end up in a NOT FOUND page as you can see below: 
 ![delete_exercise](https://user-images.githubusercontent.com/62517353/219964160-7fa77fcd-719f-4b8c-8b79-7ccd6f31adac.png)

Helmphlyx committed:
handle casing when querying for exercises

Helmphlyx committed:
fix delete queries

Helmphlyx commented:
> I liked what you did with the SQL, it's more testable and clear now. I was testing the app manually and found some inconsistencies.
> 
> 1. When searching for an exercise name, if the exercise was added by us, it sanitizes the input and finds the exercise. However, if I search for exercises names that came from the API, I need to type the capital letters or lower letters exactly like in the API. It would be helpful to sanitize the names that come from the API if possible.
> 2. If I press the delete button in the workout and the exercise, it is not actually deleting them, they still appear in the search page.
> 3. When editing a workout, if I delete an exercise, I end up in a NOT FOUND page as you can see below:
>    ![delete_exercise](https://user-images.githubusercontent.com/62517353/219964160-7fa77fcd-719f-4b8c-8b79-7ccd6f31adac.png)

Great catches!

Instead of altering the input I added a query change so that the casing stays the same, but we change the casing during query as to ignore it.

Also fixed the delete errors, caused from mixed table names.

Helmphlyx committed:
Merge branch 'main' into feature/EDAP-0000/sql_optimization
